src/main/java/address/browser/HyperBrowser.java

```diff
@@ -0,0 +1,215 @@
+package address.browser;
+
+import address.browser.page.Page;
+import address.exceptions.IllegalArgumentSizeException;
+import address.util.FxViewUtil;
+import address.util.UrlUtil;
+import com.teamdev.jxbrowser.chromium.Browser;
+import javafx.scene.Node;
+import javafx.scene.layout.AnchorPane;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A HyperBrowser capable of displaying web pages instantaneously.
+ */
+public class HyperBrowser {
```

Comment:
Implementation classes should briefly explain the mechanism (how it works), 
eg java's `ArrayList` explains that is uses a resizable random access array and explains the implications

Definitely talk about it using a cached pool of pages, and how it doesnt remember which page is visible